### PR TITLE
Adds a Station Time different from the Current Time

### PR DESCRIPTION
**Pull request by FlattestGuitar**
:cl: FlattestGuitar
tweak: The station is no longer suspended in one point in time. Look at your watch!
/:cl:

Okay, this is something I've been considering for a while, but only now have gotten around to trying to code it. It was laughably easy to make work, thanks to @TheDZD.

There are now two times shown on the status tab. First one refers to GMT ~~+1~~ time of the REAL WORLD, but also the time of the ss13 universe. The second one shows you how much time there has been since the beginning of the shift.
Really not much of a functional change in any way, since nothing really depends on time on Paradise, but it will be applicable during RP.

nobody reads this

I am 99% sure this doesn't break anything.

Also, considering all shifts begin at 12:03 with the current code this makes it absolutely possible to be correct about the current time being HIGH NOON.

*I'm sorry*

### CODE REVIEW

**Comment by Spacemanspark** [comment]
Horrible joke :-1:

**Comment by TheDZD** [comment]
High noon you say?

**Comment by taukausanake** [comment]
How dare you make me read a 24 hour clock. I want my 12 hour AM and PM cycles.

I like the possibility of an actual "real time" in SS13 but I don't feel this way is adding anything to it. Nothing you can relate to RP because the station literally blew up just a few minutes ago the previous round. Or it got turned into a cult infested nightmare after Nar-Sie humped everything. I would more care for an in-game time that stays to only itself and changes depending on what has happened the previous round. If the station blew up then fast forward time into the future to it being reconstructed. Shift went by fine (or was extended without adminbus) then the time moves ahead very little if at all while the new shift moves in. This isn't doing much of anything in those regards and less for anything else I feel, but that's just me

**Comment by VampyrBytes** [comment]
> GMT+1

There's daylight saving time in space? :wink:

**Comment by KasparoVy** [comment]
Wait, you mean the whole world isn't on eastern standard time!!?!??!

**Comment by Necaladun** [comment]
...finally I don't have to say "Good afternoon" every shift. That's...really the only effect I see of it.

Also screw working out orbital rotations and the flow of time and relativity and all that shit I have enough of that in my GURPS game atm tyvm.

**Comment by VampyrBytes** [comment]
One point, someone (guessing @tigercat2000 ) will have to update discord's :eyes:status function to work with this

**Comment by FlattestGuitar** [comment]
@taukausanake, are you implying NT *cannot* blow up the old cyberiad, bluespace in a new one and clone all crew withtin 3 minutes? Heresy.

Also, I didn't occur to me you peasants might use 12-hour clocks, I'm a dirty pole. I did some hardcore math to make it work with am/pm and it's way nicer now. I'll go nom on some kabanosy.

@VampyrBytes, the "shift time" stays unchanged, only the proc responsible for printing it to the status screen now does a `-12`, so Cyb can sleep soundly tonight.

Also, I changed it to GMT, because there's too much hassle with wrapping the hour around midnight and I don't see the point in that anyway. It's not like we're trying to cater to a specific area players might come from.

**Comment by Necaladun** [comment]
GMT = Galactic Mean Time.

**Comment by VampyrBytes** [comment]
> the "shift time" stays unchanged, only the proc responsible for printing it to the status screen now does a -12, so Cyb can sleep soundly tonight.

Yeah, I realised that. I just meant that cyb will be showing a time that doesn't actually correspond to anything visible on the server, so for example, the server will show shift time = 0:34 and cyb will be showing that it's 12:34 - I'm assuming cyb doesn't have a mob hidden away somewhere and so can't access /mob/proc/show_stat_station_time() :wink: 

It's not going to break cyb, it would just be better to have it show what is shown on the server

**Comment by Necaladun** [comment]
> I'm assuming cyb doesn't have a mob hidden away somewhere

I've stopped asking what Tiger has hidden away. You don't wanna know really.

**Comment by FlattestGuitar** [comment]
Actually, you're right. Boop. Let's bother tiger with senseless @tigercat2000 pings.

**Comment by tigercat2000** [comment]
if i have to modify cyb i'm going to :hocho: @FlattestGuitar

**Comment by FlattestGuitar** [comment]
Just add a `-12` in one spot and you're good to go.

**Comment by tigercat2000** [comment]
but really though

i don't quite understand what this does
what's the problem with having generic "station time" which starts at 12:00

**Comment by Fox-McCloud** [comment]
>what's the problem with having generic "station time" which starts at 12:00

This is kinda my thoughts, too, I mean. I'm not "argh blargh" on this, it's just...weird, I guess; the 12:00 universal reference is just kinda "one of those things". Again, I"m not entirely opposed to this, but I don't exactly see the point, either.

**Comment by Spacemanspark** [comment]
I do like it, if I didn't make my opinion clear.

**Comment by Krausus** [comment]
I disagree with the people saying this seems pointless. It's actually *worse* than pointless: by not touching `worldtime2text`, various things in-game will still generate timestamps that start at 12:00, and those timestamps won't match up with *either* of the times shown in the status tab. Also, the "shift time" will still usually be 3 minutes off, because it's still based on when the world started, rather than when the shift actually started.

As I recall, Baystation solved the always-start-at-noon problem by replacing that +12 with one of a few random offsets. We never bothered porting that, because *whoooo cares.*

**Comment by Lady-Luck** [comment]
I like it.

**Comment by Ar3nn** [comment]
I like the idea, and the only valid complaint against it is the extent to which you have to actually mess with the code to get it to work. A surprising amount of stuff references world-time, and if you don't add the time-shift to all of it, it'll still be printing 12:xx when it gets called. This might also break admin logs, if done incredibly badly, though I dunno to what extent.

**Comment by FlattestGuitar** [comment]
Mhm, i browsed through the code and it really is referenced in a lot of places. Will fix when I can.

**Comment by Krausus** [comment]
> This might also break admin logs

Admin logs use real-world time.

> Mhm, i browsed through the code and it really is referenced in a lot of places. Will fix when I can.

... please tell me you don't intend to fix this by replacing every use of `worldtime2text`.

**Comment by FlattestGuitar** [comment]
Uh, not really. Should just be possible by taking the 12 off from the proc.

**Comment by tigercat2000** [comment]
@Ar3nn there isn't a valid reason *for* it either as far as I can see
still have no idea why this is useful at all over the normal station time

**Comment by Spacemanspark** [comment]
Nothing really against it either though.

**Comment by tigercat2000** [comment]
what's against it?
 - tying real world time to the game for some reason
 - needing to change a fair amount of code for worldtime2text
 - making me need to modify cyberiadAI
 - changing how station time has always been, people get used to reading how long the round has been going from 12:00

**Comment by Regen1** [comment]
:-1: I like how it currently is

**Comment by FlattestGuitar** [comment]
Alright, I've looked through the code and found that A LOT of things use the station time in ways I didn't expect. Since my snowflaky code that would solve the emerging problems would probably not play too well with the codebase, I'll be closing this.

Sorry to disappoint.

**Comment by LittleBigKid2000** [comment]
@taukausanake
Centcom has a bluespace rapid construction device array. It can build and repair stations from lightyears away and is fast enough to rebuild an entire station in about 3 minutes. Centcom also has a copy of every crewmember in storage at all times. The copy's mind is completely blank, lacking any consciousness. When the shift ends, any dead crew members have their consciousness transferred to the copy's brain.
TL;DR: BLUESPESS

**Comment by taukausanake** [comment]
@LittleBigKid2000 Get out :P
